components/aws/sagemaker/batch_transform/src/sagemaker_transform_component.py

```diff
@@ -0,0 +1,168 @@
+# limitations under the License.
+
+import logging
+from typing import Dict
+
+from batch_transform.src.sagemaker_transform_spec import (
+    SageMakerTransformSpec,
+    SageMakerTransformInputs,
+    SageMakerTransformOutputs,
+)
+from common.sagemaker_component import (
+    SageMakerComponent,
+    ComponentMetadata,
+    SageMakerJobStatus,
+)
+
+
+@ComponentMetadata(
+    name="SageMaker - Batch Transformation",
+    description="Batch Transformation Jobs in SageMaker",
+    spec=SageMakerTransformSpec,
+)
+class SageMakerTransformComponent(SageMakerComponent):
+    """SageMaker component for transform."""
+
+    def Do(self, spec: SageMakerTransformSpec):
+        self._transform_job_name = (
+            spec.inputs.job_name
+            if spec.inputs.job_name
+            else "BatchTransform"
+            + spec.inputs.model_name[spec.inputs.model_name.index("-") :]
+        )
+        super().Do(spec.inputs, spec.outputs, spec.output_paths)
+
+    def _get_job_status(self) -> SageMakerJobStatus:
+        response = self._sm_client.describe_transform_job(
+            TransformJobName=self._transform_job_name
+        )
+        status = response["TransformJobStatus"]
+
+        if status == "Completed":
+            return SageMakerJobStatus(
+                is_completed=True, has_error=False, raw_status=status
+            )
+        if status == "Failed":
+            message = response["FailureReason"]
+            return SageMakerJobStatus(
+                is_completed=True,
+                has_error=True,
+                error_message=message,
+                raw_status=status,
+            )
+
+        return SageMakerJobStatus(is_completed=False, raw_status=status)
+
+    def _after_job_complete(
+        self,
+        job: object,
+        request: Dict,
+        inputs: SageMakerTransformInputs,
+        outputs: SageMakerTransformOutputs,
+    ):
+        outputs.output_location = inputs.output_location
+
+    def _on_job_terminated(self):
+        self._sm_client.stop_transform_job(TransformJobName=self._transform_job_name)
+
+    def _create_job_request(
+        self, inputs: SageMakerTransformInputs, outputs: SageMakerTransformOutputs,
+    ) -> Dict:
+        ### Documentation: https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/sagemaker.html#SageMaker.Client.create_transform_job
+        request = self._get_request_template("transform")
+
+        request["TransformJobName"] = self._transform_job_name
+        request["ModelName"] = inputs.model_name
+
+        if inputs.max_concurrent:
+            request["MaxConcurrentTransforms"] = inputs.max_concurrent
+
+        if inputs.max_payload or inputs.max_payload == 0:
+            request["MaxPayloadInMB"] = inputs.max_payload
+
+        if inputs.batch_strategy:
+            request["BatchStrategy"] = inputs.batch_strategy
+        else:
+            request.pop("BatchStrategy")
+
+        request["Environment"] = inputs.environment
```

Comment:
interesting doesnt check for existence

Reply:
Default is `{}`, so this is fine. But I'll add a null check just in case this changes.